% TWO COLUMNS STANDARD LAYOUT
%\documentclass[aps,pre,twocolumn]{revtex4}
% ONE COLUMN NUMBERED LAYOUT
\documentclass[aps,rmp,onecolumn,linenumbers]{revtex4-1}
\usepackage[english]{babel}
\usepackage[utf8x]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amsfonts,amssymb,eucal,eurosym,textcomp}
\usepackage{color}
\usepackage{array}
\usepackage{graphicx}
\usepackage{natbib}
\usepackage{rotating}
\graphicspath{{./figures/}}
\newcommand{\EQ}[1]{Eq.~(\ref{eq:#1})}
\newcommand{\EQS}[2]{Eqs.~(\ref{eq:#1)} and (\ref{eq:#2})}
\newcommand{\FIG}[1]{Fig.~\ref{fig:#1}}
\newcommand{\TAB}[1]{Tab.~\ref{tab:#1}}
\newcommand{\REF}[1]{ref.~\cite{#1}}
\newcommand{\SNP}{SNP}
\newcommand{\C}{$^{circ}$C}
\newcommand{\MM}{Materials and Methods}
\begin{document}
\title{Establishment and stability of the latent HIV-1 DNA reservoir}
\author{Johanna Brodin$^{1}$, Fabio Zanini$^{2,3}$, Lina Thebo$^{1}$, Christa Lanz$^{2}$, G\"oran Bratt$^4$, Richard A.~Neher$^{2}$, Jan Albert$^{1,5}$}
\affiliation{$^{1}$Department of Microbiology, Tumor and Cell Biology, Karolinska Institute, Stockholm, Sweden\\
$^{2}$Max Planck Institute for Developmental Biology, 72076 T\"ubingen, Germany\\
$^{3}$current address: Stanford University, Stanford, CA, USA\\
$^4$Department of Clinical Science and Education, Venh\"alsan, Stockholm South General Hospital, Stockholm, Sweden\\
$^5$Department of Clinical Microbiology, Karolinska University Hospital, Stockholm, Sweden
}
\date{\today}

\begin{abstract}
HIV-1 infection cannot be cured because the virus persists as integrated proviral DNA in long-lived cells despite years of suppressive antiretroviral therapy (ART). We characterized the establishment, turnover, and evolution of viral DNA reservoirs in 10 patients after 3--18 years of suppressive ART. We previously studied HIV-1 evolution in the same patients prior to therapy (Zanini et al., 2016). Comparison of DNA sequences obtained after therapy to RNA sequences obtained before therapy indicated no evolution during suppressive ART. Most of the reservoir DNA sequences were very similar to sequences of viruses that replicated in the plasma during the last year before the start of ART. The results do not support persistent HIV-1 replication as a mechanism to maintain HIV-1 reservoirs during long-term suppressive therapy. Instead, they suggest that DNA variants were turning over as long as the patients were untreated and that suppressive ART halted this turnover.
\end{abstract}

\maketitle
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% main text of the manuscript
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Introduction}
Combination antiretroviral therapy (ART) has had significant effects on the morbidity and mortality associated with human immunodeficiency virus type 1 (HIV-1) infection. ART very effectively suppresses active virus replication, but it cannot eradicate the infection because HIV-1 persists as integrated proviral DNA in long-lived cells that constitute a virus reservoir. Latently infected resting memory CD4+ T-lymphocytes (memory CD4 cells) represent the most solidly documented HIV-1 reservoir \cite{eriksson_comparative_2013,chun_quantification_1997,chun_vivo_1995}. Fully functional integrated HIV-1 proviruses are present in a small fraction of memory CD4 cells. These cells do not produce virus when they are in a resting state, but can be induced to produce virus upon activation in vitro and in vivo \cite{eriksson_comparative_2013,chun_vivo_1995,chun_quantification_1997,massanella_measuring_2016}. The activation state of the infected cell and the viral encoded Tat feedback loop jointly determine latency and virus production \cite{Razooky_Cell_2015,Rouzine_Cell_2015}.

Because of their importance to development of a cure for HIV-1 infection, many methods to quantify HIV-1 reservoirs have been developed. The quantitative virus outgrowth assay (QVOA) has been the ``gold standard'' \cite{stockenstrom_longitudinal_2015,massanella_measuring_2016,bruner_towards_2015}, but recent studies have revealed that this assay greatly underestimates the true size of the functional reservoir \cite{ho_replication-competent_2013,Bruner_Nat_Med_2016}. 
In contrast, PCR-based assays overestimate the size of the functional reservoir because they cannot distinguish between replication-competent and defective viral genomes. \cite{stockenstrom_longitudinal_2015,massanella_measuring_2016,bruner_towards_2015,Bruner_Nat_Med_2016}. Many defective proviruses contain large internal deletions \cite{ho_replication-competent_2013,sanchez_accumulation_1997}. Defective proviruses also result from APOBEC editing, which induces G-to-A hypermutation \cite{yu_single-strand_2004,kieffer_g-->hypermutation_2005,stopak_hiv-1_2003,Bruner_Nat_Med_2016}.

The HIV-1 reservoir is established early during primary infection and is remarkably quantitatively and qualitatively stable.  \citet{siliciano_long-term_2003} found a 44-month half-life for latently infected cells capable of producing replication-competent virus in the QVOA. Similarly, HIV-1 DNA levels and genetic compositions are very stable in patients receiving long-term suppressive ART \cite{stockenstrom_longitudinal_2015,besson_hiv-1_2014,josefsson_hiv-1_2013,kearney_lack_2014,gunthard_evolution_1999,evering_absence_2012,kieffer_absence_2004}. 
Early ART reduces the reservoir’s size and genetic complexity \cite{chomont_hiv_2009,josefsson_hiv-1_2013,lori_treatment_1999,strain_effect_2005}.
Most studies suggest that the HIV-1 reservoir is maintained by the physiological homeostasis of CD4 memory that in part involves occasional expansion and contraction of individual CD4 cell clones \cite{stockenstrom_longitudinal_2015,chomont_maintenance_2011,chomont_hiv_2009}. However, the results of some studies have suggested that persistent virus replication may be an important contributor to the maintenance of the HIV-1 reservoir \cite{buzon_hiv-1_2010,yukl_effect_2010}. Recently, \citet{lorenzo-redondo_persistent_2016} reported evidence of rapid HIV-1 evolution in lymphoid tissue reservoirs. 

Despite their significance for HIV-1 cure efforts, relatively little is known about the pre-ART establishment and turnover of the HIV-1 reservoir. In this study, we characterized the establishment and maintenance of the HIV-1 DNA reservoirs in 10 patients. We previously studied the evolution of HIV-1 in these patients before ART by whole genome deep-sequencing of HIV-1 RNA in longitudinal plasma samples \cite{zanini_population_2016}. We now sequenced HIV-1 DNA from peripheral blood mononuclear cells (PBMCs) from these patients after many years of suppressive ART and compared these reservoir DNA sequences with the replicating HIV populations that were present before ART. The collection dates of all available samples relative to start of treatment are presented in \FIG{study_design}. 

We found that the HIV-1 DNA populations remained genetically stable for up to 18 years after the start of suppressive ART. The absence of genetic changes indicates that viral evolution and replication are not important mechanisms for the maintenance of HIV-1 reservoirs during supressive ART. We also found that the variants that were replicating shortly before start of ART were overrepresented in the HIV-1 DNA reservoirs. This excess of late variants in the DNA reservoirs indicated that proviral HIV-1 variants continued to turn over with a half-life of approximately one year until the patients began therapy. ART effectively froze the composition of the HIV-1 DNA reservoir in the state it had at start of therapy. 

\section*{Results}

\subsection*{Patients and samples} 
The study population consisted of 10 HIV-1 infected patients who were diagnosed in Sweden between 1990 and 2003. The following criteria were used to select the patients: 1) A relatively well-defined time of infection; 2) Being treatment-naive for $\geq 5$ years; and 3) Receiving suppressive ART (plasma HIV-1 RNA levels continuously \textless 50 copies/ml) for $≥2$ years. 
We previously characterized HIV-1 RNA populations in longitudinal plasma samples (6--12 per patients over 5–8 years). 
Results for 9 out of the 10 patients were reported in \cite{zanini_population_2016}
The tenth patient (p7) was part of the previous study, but was not included in the final analyses of that study because the sequencing of plasma samples obtained during the first years after infection had failed. The results for the patient characteristics are presented in \FIG{study_design} and Table 1. 

For this study, we obtained sequence data from the HIV-1 DNA in viral reservoirs using deep sequencing of the p17gag region of the HIV-1 genome in DNA prepared from PBMCs. Longitudinal PBMC samples (1--3 samples per patient for up to a 2.6-year period) were obtained 3--18 years after the start of suppressive ART (\FIG{study_design} and Table 1). We defined viral DNA reservoirs as HIV-1 p17gag sequences that were still present in PBMCs after at least 2 years of suppressive ART. HIV-1 DNA template numbers were quantified by limiting dilution by the same p17gag PCR that was used for sequencing. Identical p17gag sequences were merged into haplotypes while preserving their abundance. Minor haplotypes were merged with major haplotypes if they differed by one mutation (see Materials and Methods section). Processed sequence data are available at \texttt{hiv.tuebingen.mpg.de}. Raw sequencing reads from all HIV-1 DNA samples were deposited in the European Nucleotide archive (study accession number PRJEB13841; sample accession numbers ERS1138001-ERS1138025).

\subsection*{Proviral DNA sequences reflect pretreatment RNA sequences}
The HIV-1 DNA sequences recapitulate the diversity observed in RNA sequences before treatment, often with exact sequence matches, \FIG{tree_example} and Figure 2 -- supplement 1. While we observed large variations in the abundance of haplotypes with sequence read frequencies varying between 0.1\% and 50\% (Figure 2 -- supplement 2), the high similarities between RNA and DNA sequences confirmed that our characterization of proviral diversity was specific and sensitive. 
Variation in haplotype abundance likely reflects clonal expansion \cite{josefsson_hiv-1_2013,stockenstrom_longitudinal_2015}, independent integration of identical sequences, and resampling of the same original DNA templates during sequencing. However, the specific contributions of these distinct mechanisms were difficult to determine in our sequencing results. 

The estimated numbers of HIV DNA templates, the numbers of distinct haplotypes observed, and the fractions of haplotypes present in multiple samples are presented in Table S1. If a haplotype was present at a frequency $>1$\%, it was also present in another sample from the same patient in approximately one-third (median 0.29) of all cases.

\subsection*{Hypermutated sequences are frequent in HIV-1 reservoirs}
We found that substantial proportions (median 14\%; range 0--42\%) of the p17gag DNA sequences from the viral reservoirs were hypermutated and were therefore expected to be replication incompetent (Figure 2 -- supplement 3). This result was consistent with the results of earlier research, which showed that 9--30\% of sequences were hypermutated \cite{josefsson_hiv-1_2013,bruner_towards_2015,kieffer_g-->hypermutation_2005}. A small proportion of sequences had stop codons that were not obviously due to G-to-A hypermutation (mean 3\%, range 0--12\%). A proportion of sequences without overt inactivating mutations were likely also replication incompetent due to mutations or deletions outside of p17gag \cite{ho_replication-competent_2013,Bruner_Nat_Med_2016}. 
We excluded hypermutated sequences from the main analyses, but we also performed complementary analyses that included hypermutated sequences. 

\subsection*{Lack of evidence of persistent replication in HIV-1 DNA reservoirs}
Whether or not HIV-1 reservoirs are maintained by persistent replication remains controversial \cite{stockenstrom_longitudinal_2015,chomont_maintenance_2011,chomont_hiv_2009,buzon_hiv-1_2010,yukl_effect_2010,lorenzo-redondo_persistent_2016,evering_absence_2012}. We used the p17gag DNA sequences from viral reservoirs to search for evidence of sequence evolution, which should occur if the virus was replicating. Root-to-tip distances for plasma RNA populations and PBMC DNA populations were calculated relative to the major RNA haplotype present in the first plasma sample.

\FIG{root_to_tip} presents the results for temporal changes in root-to-tip distances in HIV-1 RNA and DNA populations obtained before and after the start of suppressive ART, respectively. Plasma HIV-1 RNA populations obtained before the start of ART evolved at a relatively constant rate \cite{zanini_population_2016}; there was a steady increase in mean root-to-tip distances over time in \FIG{root_to_tip}. In sharp contrast, the HIV-1 DNA populations obtained after 3--18 years of suppressive therapy had stable root-to-tip distances. Hypermutated DNA sequences had larger root-to-tip distances, but these distances were also stable over time (Figure 3 -- supplement 1). To rule out biases due to clonal expansion or PCR resampling, or both, we repeated this analyses while counting each unique sequence only once; we obtained essentially the same results (Figure 3 - supplement 2).

The results for the rates of evolution before and after the start of suppressive ART are presented in Table 2. In all 10 patients we found statistically significant evolution of plasma RNA sequences before the start of therapy, with rates $1-4\times 10^{-3}$/year. In contrast, DNA sequences showed no signal of statistically significant evolution in DNA reservoirs during suppressive ART. 

Taken together, our results did not indicate that persistent HIV-1 replication acts as a mechanism to maintain the HIV-1 reservoir during suppressive therapy.

\subsection*{Time for deposition of reservoir HIV-1 DNA sequences}
The phylogenetic analyses indicated that most of the HIV-1 DNA variants present in the PBMCs matched the HIV-1 RNA variants detected in the plasma samples obtained shortly before the start of therapy (\FIG{tree_example} and Figure 2 -- supplement 1). However, the phylogenetic trees also revealed that DNA variants that matched the earliest plasma variants were present in some patients. 

To investigate when the PBMC HIV-1 DNA variants were deposited in the viral reservoirs, we compared the on-treatment PBMC DNA sequences with the longitudinal pre-treatment plasma RNA sequences. For each p17gag DNA sequence, we determined the pre-treatment plasma sample and the RNA haplotype that was the most likely source. 
This method assigned most of the HIV-1 DNA sequences to the plasma samples closest to the start of treatment (\FIG{seeding}, panel A and Figure 4 -- supplement 1). 
A combined analysis of the data for all patients (\FIG{seeding}, panel C) indicated that approximately 60\% of the DNA reads were most similar to RNA variants that were present in plasma samples obtained during the last year before the start of therapy. The representation of the variants present in the earlier plasma samples decreased; the half-life was 0.75 years going backward in time from the start of therapy. Analyses in which each unique sequence was counted only once revealed very similar results, which indicated that the findings were robust to possible sampling biases from clonal expansions or PCR resampling, or both (Figure 4 -- supplement 2).

Sequences that matched sequences at earlier plasma sampling time points were found as minor variants among the p17gag DNA sequences (\FIG{seeding}, panel B). Among these minor variants, the DNA sequences matching plasma variants obtained within 6 months post-infection were overrepresented in some patients (14\%, 2.4\%, 42\%, \textless 1\%, and 6.9\% of all the reads in patients 2, 3, 6, 8, and 11, respectively). 
On average approximately 5\% of the reads matched plasma variants from the first 6 months after the estimated date of infection (EDI) (\FIG{seeding}, panel B). If the outlying data for patient 6 were omitted, the this fraction was 2\%. Even 2\% was unexpectedly high given the rapid DNA decay rate present during untreated HIV-1 infection. If the rates of seeding into the reservoirs were stable over time, we would have expected a mean value of 0.14\% (range \textless 0.001–0.32\% across patients) of the DNA sequences to match the earliest plasma variants. This result suggested that massive seeding of HIV DNA into the viral reservoirs occurred during the first weeks and months post-infection. 

In summary, viral variants that replicated shortly before the start of suppressive ART were overrepresented in the HIV-1 DNA reservoirs. This result indicated that the infected cells were turning over (approximately 1-year half-life) for as long as the patients were untreated. Suppressive ART halted this turnover. 

\section*{Discussion}
We investigated the composition and turnover of HIV-1 DNA sequences in viral reservoirs in patients receiving long-term suppressive therapy. The reservoir HIV-1 DNA populations were remarkably stable and showed no evidence of active replication during ART. Since we had previously characterized the HIV-1 populations in these patients prior to ART, we could determined how sequences in the HIV-1 reservoir related to pre-treatment populations. In particular, we were able to show that sequences in HIV-1 DNA reservoirs mainly derived from the populations present during the last year before the start of suppressive therapy.

Our results indicate that persistent HIV-1 replication is not a mechanism for maintenance of HIV-1 reservoirs during suppressive therapy (\FIG{root_to_tip} and Table 2). This conclusion differs from the results of a recent report by \citet{lorenzo-redondo_persistent_2016} and of a few earlier reports \cite{yukl_effect_2010,buzon_hiv-1_2010}. However, our results are consistent with the results of other earlier studies \cite{stockenstrom_longitudinal_2015,besson_hiv-1_2014,josefsson_hiv-1_2013,kearney_lack_2014,gunthard_evolution_1999,evering_absence_2012,kieffer_absence_2004}. \citet{lorenzo-redondo_persistent_2016} compared genetic diversity in HIV-1 RNA in plasma samples at the start of therapy with HIV-1 DNA sequences obtained from blood and tissues at baseline, 3 months, and 6 months after the start of treatment. The authors reported very high rates of evolution ($7.4-12\times 10^{-3}$ changes per site per year); these rates are approximately 3- to 5-fold greater than those typically observed in gag and pol of replicating RNA populations. 
Such rapid evolution is incompatible with the lack of observable changes in the reservoir sequences over the 20 times longer time intervals reported here (Figure 3 -- supplement 2).

Without longitudinal data on the evolution of HIV-1 populations prior to treatment, the nature of the changes reported by \citet{lorenzo-redondo_persistent_2016} are difficult to determine. It is possible that the reported temporal signal arises from changes in DNA reservoir composition during the first 6 months of therapy when short-lived cells infected with recently replicating virus variants gradually disappear. Death of short-lived cells increases the proportions of longer-lived cells that sample further back into the history of the infection. This scenario (Figure 3 -- supplement 3) would result in sequence changes that do not indicate (forward) evolution. Instead these changes can create an impression of ``backward'' evolution towards older HIV variants due the preferential pruning of later variants. 

\citet{lorenzo-redondo_persistent_2016} investigated HIV-1 DNA sequences in tissue and PBMC samples, but we only examined PBMC samples. However, Lorenzo-Redondo et al.~found similar rates in PBMC compared to tissue samples. Furthermore, tissue and blood HIV-1 DNA variants should be well-mixed during the time period that we investigated \cite{josefsson_hiv-1_2013,stockenstrom_longitudinal_2015,lorenzo-redondo_persistent_2016}. 

Both \citet{lorenzo-redondo_persistent_2016} and we studied DNA sequences in HIV-1 reservoirs, which contain high proportions of defective virus \cite{ho_replication-competent_2013,Bruner_Nat_Med_2016}. The absence of provirus evolution or turnover does not fully exclude the possibility that there was replication and evolution of replication-competent viruses present in the reservoirs. However, if such replication occurred, it involved a minority of infected cells and occurred below the detection limit of the deep-sequencing method. To enrich for replication-competent and putatively evolving virus, QVOA followed by sequencing of virus released into the supernatant should be performed, not sequencing the total HIV-1 DNA (as done by \citet{lorenzo-redondo_persistent_2016}, us, and others \cite{josefsson_hiv-1_2013,kieffer_g-->hypermutation_2005,evering_absence_2012}).
Our finding of genetic stability in the DNA reservoirs is consistent with \citet{josefsson_hiv-1_2013} and \citet{stockenstrom_longitudinal_2015}. They found that defective HIV-1 DNA integrants present during long-term effective ART appear to be maintained by proliferation and longevity of infected cells and not by ongoing viral replication. 

We had detailed longitudinal data on the evolution of the plasma HIV-1 RNA population from the time of infection to the start of suppressive ART. Therefore, we could determine the time at which the viruses in the DNA reservoirs had been deposited (\FIG{seeding} and supplements). We found that a majority of the variants present in the HIV-1 DNA reservoirs were derived from HIV-1 RNA variants that had replicated during the last year before the start of suppressive ART. In contrast, \citet{frenkel_multiple_2003} reported the persistence of greater numbers of earlier versus more recent virus variants in a few children receiving suppressive ART. We found a similar, but much less prominent, presence of variants replicating during the first 6 months post-infection (\FIG{seeding}, panel B). These early variants were overrepresented relative to the expected level if deposition had been uniform over time. This overrepresentation could be due to the high viral load present during primary HIV-1 infection and is consistent with results by \citet{Bruner_Nat_Med_2016}, who found that defective proviruses rapidly accumulate during acute HIV-1 infection. However, we observed such an excess of early variants in only 5 out 10 patients. 

Defective HIV-1 proviruses are unique in vivo labels of individual memory CD4 cell clones. Similar to the sequencing of T-cell receptors, these labels can be used to track the fates of the memory CD4 cells \cite{robins_immunosequencing:_2013}. This strategy was used by \citet{imamichi_lifespan_2014}, who found that a T-cell clone carrying a defective HIV-1 provirus can persist for $>17$ years. Similarly, prenatally formed T-cell receptors shared by twins have lifetimes that are $>30$ years \cite{pogorelyy_persisting_2016}. 
Our results are consistent with such long T-cell life times once HIV replication is suppressed by ART.
In untreated infection, however, we estimate much faster turnover of infected cells with a half-life of approximately 1 year. 
This conclusion is based on the observation that most HIV-1 DNA sequences derive from replicating virus shortly before the start of ART. 
The results of earlier studies, which were based on different types of CD4 cell labelling, have indicated that CD4 cell die at a 3- to 4-fold increased rate in untreated HIV-1-infected patients, compared with uninfected individuals and with patients receiving suppressive ART \cite{hellerstein_directly_1999,mccune_factors_2000,ribeiro_vivo_2002}. The more dramatic difference that we observed might be due to that earlier studies estimated lifespans of individual cells, whereas we estimated the lifespans of CD4 cell clones carrying defective proviruses (i.e., infected cells and their daughter cells). 

Our study had several limitations. We did not sort cells and therefore could not investigate whether there were differences in HIV-1 turnover rates between different types and subsets of cells (e.g., memory CD4 cells and their subsets). However, it is reasonable to assume that most of our HIV-1 DNA sequences were from memory CD4+ T-lymphocytes because others have found that these cells are the main HIV-1 reservoir \cite{eriksson_comparative_2013,chun_quantification_1997,chun_vivo_1995}. 
Because we sequenced a relatively short region of the HIV-1 genome, we could not reliably distinguish between replication-competent and defective viruses. We did not find evidence of evolution in proviral DNA sequences, but we cannot rule out the possibility that a small subset of viruses was replicating and remained undetected among the many defective viruses. However, our analyses of the composition and turnover of the virus reservoir were aided by the fact that most of the viruses were defective because these proviruses acted as inert in vivo labels of CD4 cell clones.
We observed large variations in the abundance of sequence haplotypes that likely indicate the presence of clonal expansions \cite{josefsson_hiv-1_2013,stockenstrom_longitudinal_2015}, independent integrations of identical sequences, and resampling of the same original DNA templates during sequencing. 
Given our sequencing method, we could not determine the relative contributions of these mechanisms. We are using Primer ID sequencing \cite{jabara_accurate_2011} to better understand the in vivo dynamics of different viral haplotypes. Analysis of HIV integration sites represents an alternative method of clonal expansion identification \cite{maldarelli_specific_2014,cohn_hiv-1_2015}. However, such analysis was not possible for us because we did not sequence the extreme ends of the LTRs. We analyzed p17gag because it was part of our plasma HIV RNA dataset and because it has sufficient genetic diversity to allow for accurate and valid comparisons between plasma RNA and PBMC DNA HIV populations. Results from several analyses suggested that clonal expansion did not affect our results in a qualitative manner. Clonal expansion would have resulted in artefactual over-representation of some sequences. To control for this, we repeated the analyses in Figures 3 and 4 and counted each haplotype only once. We obtained very similar results (Figure 3 -- supplement 2 and Figure 4 -- supplement 2). Consistent with our results and interpretation, \citet{cohn_hiv-1_2015} found only a limited decrease in single integrations following the start of ART (from approximately 70\% to 50\%).

In summary, we found compelling evidence against persistent viral replication as a mechanism to maintain the latent HIV-1 DNA reservoir during suppressive therapy. We also found that most of the latently infected cells present during long-term suppressive ART were infected shortly before the start of ART, and that T-cell turnover slowed down dramatically when ART began.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% materials and methods section
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Materials and methods}

\paragraph*{Ethical statement}
The study was performed according to Declaration of Helsinki requirements. Ethical approval was granted by the Regional Ethical Review Board, Stockholm, Sweden (Dnr 2012/505 and 2014/646). Written and oral informed consent was obtained from each patient that participated in the study.

\paragraph*{Patients}
The study population consisted of 10 HIV-1-infected patients who were diagnosed in Sweden between 1990 and 2003. These patients had been included in a recent study of the population genomics of intrapatient HIV-1 evolution \cite{zanini_population_2016}. The patients were selected based on the following inclusion criteria: 1) A relatively well-defined time of infection (based on a negative HIV antibody test $<2$ years before a first positive test or a laboratory-documented primary HIV infection); 2) No ART during a minimum period of approximately 5 years after diagnosis; 3) Availability of biobank plasma samples covering this time period; and 4) Later initiation of successful ART (plasma viral levels $<50$ copies/ml) for $≥2$ years. As previously described, 6--12 plasma samples per patient were retrieved from biobanks and used for full-genome HIV-1 RNA sequencing \cite{zanini_population_2016}. The same patient nomenclature was used in both studies. For this study, the same 10 patients were asked to donate 70 ml fresh EDTA-treated blood on up to three occasions during a 2.5-year period. These blood samples were obtained 3--18 years after the start of successful ART. The clinical and laboratory findings (including Fiebig staging and BED testing) were used to calculate the EDI, as previously described \cite{zanini_population_2016}. The results for the characteristics of the patients and the samples are presented in Table 1. 

\paragraph*{HIV-1 RNA sequencing from plasma}
Whole-genome deep-sequencing of the virus RNA populations present in the plasma samples obtained before the start of therapy was performed as previously described \cite{zanini_population_2016}. Briefly, the total plasma RNA was extracted using an RNeasy® Lipid Tissue Mini Kit (Qiagen Cat No. 74804). The extracted RNA was amplified using a one-step RT-PCR, outer primers for six overlapping regions, and Superscript® III One-Step RT-PCR with Platinum® Taq High Fidelity High Enzyme Mix (Invitrogen, Carlsbad, California, US). An optimized Illumina Nextera XT library preparation protocol and a kit from the same supplier were used to build the DNA libraries, which were sequenced using an Illumina MiSeq instrument and 2 × 250 bp or 2 × 300 bp sequencing kits (MS-102-2003/MS-10-3003). For the present study, a section of the p17gag region of the HIV-1 genome was extracted from the full-genome RNA data set. A median number of 146 (inter-quartile range 56--400) high quality reads was obtained. The cDNA template numbers are available in \citet{zanini_population_2016}.

\paragraph*{HIV-1 DNA sequencing from PBMCs}
A total of approximately 70 ml fresh whole blood was split between seven Vacutainer (EDTA) tubes. The PMBCs were isolated using Ficoll-Paque PLUS (GE Healthcare Bio-Sciences AB, Uppsala, Sweden) centrifugation according to the manufacturer’s instructions. Total DNA was extracted from the PBMCs using the OMEGA E.Z.N.A® Blood DNA Mini Kit (Omega Bio-Tek, Norcross, Georgia) or the QIAamp DNA Blood Mini Kit (Qiagen GmbH, Hilden, Germany) according to the manufacturer’s instructions.
The amount of DNA was measured using a Qubit® dsDNA HS Assay Kit (Invitrogen™, Eugene, Oregon, USA). 
Patient-specific nested primers (Integrated DNA Technologies) were used to amplify a 387-bp long portion of the p17gag gene corresponding to positions 787 to 1173 in the HxB2 reference sequence. The primers were designed based on the plasma RNA sequences from each patient (Table S2). 
Outer primers and Platinum® Taq DNA Polymerase High Fidelity (Invitrogen™, Carlsbad, California, US) were used for the first PCR. The protocol began with a denaturation step at 94°C for 2 min followed by 15 PCR cycles of denaturation at 94°C for 20 s, annealing at 50°C for 20 s, extension at 72°C for 30 s, and a final extension step at 72°C for 6 min. For the second PCR, 2.5 μl of the product from the first PCR was amplified using inner primers and the cycle profile and enzyme as for the first PCR. Amplified DNA was purified using Agencourt AMPure XP (Beckman Coulter, Beverly, Massachusetts) and quantified using Qubit.
For each sample, the number of HIV-1 DNA templates used for sequencing was roughly quantified in triplicate by limiting dilution using the same PCR conditions, three dilutions (usually 0.5 µg, 0.1 µg, 0.02 µg DNA), and Poisson statistics.
The plasmids NL4-3 and SF162 were used for control experiments, which were performed to evaluate PCR-induced recombination. Equal proportions of the plasmids were spiked into human DNA and were amplified using the same PCR conditions as previously described. The results revealed that there was minimal PCR-induced recombination in this short amplicon. 

\paragraph*{Sequencing and read processing}
The HIV-specific primers were flanked by NexteraXT adapters. To construct sequencing libraries, indices and sequencing primers were added in 12--15 cycles of additional PCR. 
The amplicons were sequenced using an Illumina MiSeq instrument and 2 × 250 cycle kits. A total of 6,500 to 190,000 (median 35,000) paired-end reads were generated per sample. The overlapping paired-end sequencing reads were merged to create synthetic reads spanning the entire p17gag amplicon. The nucleotide on the read with the higher quality score was used for the cases of disagreement between paired reads. We counted the numbers of times a particular p17gag sequence was observed, and performed subsequent analysis using read-abundance pairs.
To reduce the effects of sequencing and PCR errors, we combined rare sequences (below frequency 0.002) with common sequences if they differed at no more than one position. Specifically, starting with the rarest sequences, we merged rare sequences with the most common sequence that was one base away. The cutoff value of 0.002 is the typical error frequency of the pipeline \cite{zanini_population_2016}. 
Less than 1 in 1000 reads that began with the HIV specific primer sequence could not be assigned with confidence to the previously determined RNA sequences. Most of these reads mapped to the human genome.
All analyses were performed using Python and the libraries NumPy, BioPython, and Matplotlib \cite{cock_biopython:_2009,van_der_walt_numpy_2011,hunter_matplotlib:_2007}.

\paragraph*{Hypermutation detection}
To classify sequences into hypermutated and non-hypermutated sequences, we counted mutations at positions that did not vary in the RNA samples obtained before therapy. If greater than four mutations were observed, and at least one-half of them were \texttt{G}$\to$\texttt{A}, the sequence was considered to be a hypermutant. The results for the distributions of the different transition mutations relative to the closest genome found in the RNA samples are presented in Figure 2 – supplement 3 for reads classified as hypermutants, or not. The results we obtained for sequences classified as non-hypermutants were very similar to the results obtained when only using sequences without stop codons.

\paragraph*{Phylogenetic analysis}
We reconstructed phylogenetic trees using the approximate maximum likelihood method implemented in the FastTree software \cite{price_fasttree_2010}. The tips were annotated with frequency, source, and sample date using custom-made Python scripts.

\paragraph*{Statistical analysis}
The root-to-tip distances were calculated as the mean distance between a sample and the founder sequence, approximated by the consensus sequence of the first RNA sample. This root-to-tip distance was regressed against time to determine the rate of evolution in the absence of treatment. The root-to-tip sequence of the last RNA sample and the DNA samples was regressed against time to determine the rate of evolution on treatment. To determine the most likely seeding time for a PBMC p17gag DNA sequence, we calculated the likelihood of sampling this sequence given the SNP frequencies in each plasma RNA sample and assigned the sequence to the sample that had the greatest likelihood value.

\section*{Acknowledgements}
%This work was supported by the European Research Council (grant Stg.~260686) and the Swedish Research Council (grant K2014-57X-09935). 
We would like to express our gratitude to the study participants.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% the following line includes the references
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\bibliography{bibliography}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% start a new page with the figures, use onecolumn layout from here
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\onecolumngrid

%%% FIGURE 1
\begin{figure*}[tb]
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig1.pdf}
\caption{Sampling times before and after start of suppressive antiretroviral therapy. For each study participant, the thick grey bar indicates the period of untreated HIV-1 replication. Circles indicate the collection times of the plasma samples used for whole genome deep sequencing of the HIV-1 RNA populations \cite{zanini_population_2016}. Triangles and squares indicate the collection times of the PBMC samples used for p17gag deep sequencing of the HIV-1 DNA populations. All times are relative to start of therapy.}
\label{fig:study_design}
\end{figure*}

%%% FIGURE 2
\begin{figure*}[tb]
\centering
\includegraphics[width=\columnwidth]{figures/fig2.pdf}
\caption{Reconstructed phylogenetic trees of plasma HIV-1 RNA sequences (circles) and PBMC HIV-1 DNA (triangles and squares) from two patients. The RNA sequences were obtained from plasma samples collected before the start of suppressive antiretroviral therapy (ART). DNA sequences 1, 2, and 3 were obtained from PBMCs collected after many years of suppressive ART (see \FIG{study_design}). The symbol colors indicate the sample date relative to the start of therapy and use the same color scale as in \FIG{study_design}. The symbol size indicates the fraction of reads represented by the node. The trees were built using the FastTree software (see Methods section) \cite{price_fasttree_2010}. Analogous phylogenetic trees for the remaining 8 study patients are presented in Figure 2 -- supplement 1.
}
\label{fig:tree_example}
\end{figure*}

%%% FIGURE 3
\begin{figure*}[tb]
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig3}
\caption{{\bf Root-to-tip distances.} The plasma HIV-1 RNA sequences evolved steadily before the start of antiretroviral therapy (ART), while no evidence of evolution was found in the PBMC HIV-1 DNA sequences obtained after the start of ART. For each patient, we have samples obtained before or at the start of therapy (HIV-1 RNA from plasma), and samples obtained after the start of therapy (HIV-1 DNA from PBMCs). The error bars indicate $\pm$ one standard deviation of the root-to-tip distances. The figure contains data on DNA sequences not classified as hypermutants. The analogous figures for the hypermutants and haplotypes are presented in Figure 3 -- supplements 1 and 2, respectively.}
\label{fig:root_to_tip}
\end{figure*}

%%% FIGURE 4
\begin{figure}[htb]
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig4}
\caption{{\bf Probable origin of sequences in the DNA reservoir.} For each HIV-1 DNA read obtained from the PBMCs, we determined the pre-treatment plasma sample and HIV-1 RNA variant that was the most likely origin of the read. Panel A presents the results for the distribution of these pre-treatment assignments for HIV-1 DNA reads from the three PBMC samples (DNA 1, 2, and 3) of patients p1 and p8 (compare trees in \FIG{tree_example}). The analogous graphs for all patients are presented in Figure 4 -- supplement 1. A large fraction of the reservoir DNA sequences was most closely related to the RNA sequences present in the plasma samples obtained during the last year before the start of therapy. 
Panels B and C present summaries of the results for the distributions of the likely origins of the reservoir reads for all samples as a function of time since estimated date of infection (EDI) and time before treatment start, respectively. The distribution of reads decays with the time from treatment start (half-life of approximately 0.75 years) (Panel C). 
A small fraction of reads was estimated to originate from plasma samples obtained shortly post-infection (Panel B). We found these early reservoir sequences in 50\% of the patients.} 
\label{fig:seeding}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% START NEW PAGE FOR THE TABLES
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
%%% PATIENT SUMMARY TABLE. TABLE 1
\begin{table*}[tb]
%\scriptsize
\begin{tabular}{|lcccc||c|cc||c|c|}
\hline
{\bf Patient} & {\bf Gender} &{\bf Transmission} & {\bf Subtype} & {\bf Age$^{a}$} & 
\multicolumn{3}{c||}{\bf HIV RNA from plasma}&
\multicolumn{2}{c|}{\bf HIV DNA from PBMCs}\\
&&&&& \#~samples& 
\multicolumn{2}{|c||}{first/last since EDI$^{b}$} & time on ART$^{b}$ & \#~templates \\
\hline
p1 & F & HET & 01\_AE &37& 12 & 0.3 & 8.2&7.9/ 9.9/ 10.4&820/ 148/ 38 \\
p2 & M & MSM &B &32& 6 & 0.2 & 5.5&6.9&75\\ 
p3 & M & MSM &B &52& 10 & 0.4 & 8.4&4.6/6.7/ 7.2&243/ 102/ 108 \\ 
p5 & M & MSM &B &38& 7 & 0.4 & 5.9&4.0/ 6.3&180/ 72 \\
p6 & M & HET &C &31& 7 & 0.2 & 7.0&3.0/ 5.0/ 5.5&115 /15/ nd \\ 
p7 & M & MSM &B &31& 11 & 6.3$^{c}$& 16.1&6.3/ 8.4/ 8.8&88/ 279/ 108 \\ 
p8 & M & MSM &B &35& 7 & 0.2 & 6.0&8.4/ 10.6/ 10.9&180/ 55/ 175 \\ 
p9 & M & MSM &B &32& 8 & 0.3 & 8.1&7.7/ 9.7/ 10.2&60/ 72/ 72 \\ 
p10 & M & MSM &B &34& 9 & 0.1 & 6.2&16.2/ 18.3/ 18.6&249/ 116/ 51 \\ 
p11 & M & MSM &B &53& 7 & 0.6 & 5.6&6.4/ 8.4/ 8.8&124/ 120/ 123 \\\hline 
\end{tabular}
\begin{flushleft}
{{\bf Table 1. Summary of patient characteristics.} $^{a}$ at diagnosis; $^{b}$ EDI: estimated date of infection; all times are given in years; $^{c}$ sequencing failed in earlier samples due to low plasma HIV-1 RNA levels.}
\end{flushleft}
\label{tab:patients}
\end{table*}
%%% TABLE WITH ESTIMATES EVOLUTIONARY RATES, TABLE 2
\begin{table}[tb]
\centering
\begin{tabular}{|c|r|r||r|r|}
\hline
{\bf patient} & \multicolumn{2}{c||}{{\bf RNA rate}} & \multicolumn{2}{c|}{{\bf DNA rate}}\\
& $[\mathrm{year}^{-1}]$& $p$-value & $[\mathrm{year}^{-1}]$& $p$-value \\\hline
p1 & $4.4\times 10^{-3}$ & $<10^{-6}$ & $-6\times 10^{-4}$ & 0.22 \\
p2 & $3.7\times 10^{-3}$ & $<10^{-2}$ & $-8\times 10^{-4}$ & -- \\
p3 & $4.1\times 10^{-3}$ & $<10^{-6}$ & $-2\times 10^{-4}$ & 0.39 \\
p5 & $4.8\times 10^{-3}$ & $<10^{-3}$ & $4\times 10^{-4}$ & 0.45 \\
p6 & $1.4\times 10^{-3}$ & $<10^{-3}$ & $-9\times 10^{-4}$ & 0.22 \\
p7 & $1.3\times 10^{-3}$ & $<10^{-2}$ & $-7\times 10^{-4}$ & 0.14 \\
p8 & $2.9\times 10^{-3}$ & $<10^{-5}$ & $8\times 10^{-5}$ & 0.22 \\
p9 & $2.6\times 10^{-3}$ & $<10^{-4}$ & $1\times 10^{-4}$ & 0.12 \\
p10 & $3.6\times 10^{-3}$ & $<10^{-5}$ & $-1\times 10^{-4}$ & 0.20 \\
p11 & $1.2\times 10^{-3}$ & $<10^{-2}$ & $2\times 10^{-4}$ & 0.16 \\\hline
\end{tabular}
\begin{flushleft}
{{\bf Table 2.} Rates of evolution in plasma HIV-1 RNA and PBMC HIV-1 DNA sequences obtained before the start and after the start of suppressive antiretroviral therapy, respectively.} 
\end{flushleft}
\label{tab:rates}
\end{table}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% SUPPLEMENTARY FIGURES AND TABLES (new page, reset table and figure counters)
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\renewcommand\thefigure{} 
\renewcommand{\figurename}{}
\setcounter{figure}{0} 
\renewcommand\thetable{S\arabic{table}} 
\setcounter{table}{0} 
%%% FIGURE 2 -- SUPPLEMENT 1
\begin{figure*}
\centering
\includegraphics[width=1.3\columnwidth,angle=90]{figures/fig2_S1}
\caption{Figure 2 -- supplement 1: Phylogenetic trees of plasma HIV-1 RNA and PBMC HIV-1 DNA sequences from all patients included in the study.}
\label{fig:supp_trees}
\end{figure*}
%%% FIGURE 2 -- SUPPLEMENT 2
\begin{figure}[hb]
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig2_S2}
\caption{Figure 2 -- supplement 2: Distributions of frequencies of haplotypes. We found a wide variation in haplotype frequencies (measured by fraction of reads), from $<0.001$ to$ >0.5$. Most of the haplotypes were present at frequencies of approximately 0.01 (note the logarithmic scales). }
\label{fig:haplo_frequencies}
\end{figure}
%%% FIGURE 2 -- SUPPLEMENT 3
\begin{figure}[hb]
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig2_S3}
\caption{Figure 2 -- supplement 3: Distributions of mutations in reads classified as hypermutanted or as non-hypermutated.}
\label{fig:hypermuts}
\end{figure}
%%% FIGURE 3 -- SUPPLEMENT 1
\clearpage
\begin{figure*}
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig3_S1}
\caption{Figure 3 -- supplement 1: Mean root-to-tip distances for plasma HIV-1 RNA sequences obtained before the start of ART and PBMC HIV-1 DNA sequences obtained after the start of ART. This figure is analogous to Figure 3 in the main text, but presents root-to-tip distance values for DNA sequences classified as hypermutants. The root-to-tip distances of the hypermutant HIV-1 DNA sequences were approximately 2\% and 4\% greater than the non-hypermutant sequences from the same samples, but the root-to-tip distances did not change over time.}
\label{fig:root_to_tip_hyper}
\end{figure*}
%%% FIGURE 3 -- SUPPLEMENT 2
\begin{figure*}
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig3_S2}
\caption{Figure 3 -- supplement 2: Mean root-to-tip distances for plasma HIV-1 RNA sequences obtained before the start of antiretroviral therapy (ART) and PBMC HIV-1 DNA sequences obtained after the start of ART. This figure is analogous to Figure 3 in the main text. Instead of weighing each sequence by its number of reads, each sequence was counted only once. The two approaches yielded similar results. The grey cone in the upper right quadrant indicates the rate of evolution during suppressive therapy estimated by \citet{lorenzo-redondo_persistent_2016}.}
\label{fig:root_to_tip_hap}
\end{figure*}
%%% FIGURE 3 -- SUPPLEMENT 3
\begin{figure}[hb]
\centering
\includegraphics[width=0.7\columnwidth]{figures/fig3_S3}
\caption{Figure 3 -- supplement 3: {\bf Short lived cells can generate a false signal of evolution.} HIV-1 DNA from cell samples taken at the initiation of antiretroviral therapy (ART) will be dominated by virus variants that have very recently infected short-lived cells (blue). The short-lived cells will die within a few months of therapy, and the HIV-1 DNA will be dominated by infected cells with longer life spans (red). The HIV-1 DNA in these cells will contain a larger fraction of sequences corresponding to virus that replicated earlier during the course of the infection. This shift in the cell populations can result in a false signal of evolution (Panel B). This illustration assumes that 90\% of the infected cells were short-lived (half-life of 21 days). The remaining 10\% of the cells were assigned a half-life of 350 days. 
To mimic the HIV-1 DNA samples obtained from the cells sampled at 0, 3, and 6 months after initiation of ART (the sampling times used by \citet{lorenzo-redondo_persistent_2016}), we randomly sampled sequences obtained from the HIV-1 RNA present before treatment (distribution illustrated in panel A). From these sequences, we calculated the \emph{apparent} divergence of the pseudo-samples at 3 and 6 months from the pseudo-sample taken when the ART was started.
There was large variation in this \emph{backward evolution} signal among patients. Averaged over patients, the observed shift corresponded to a rate of evolution of 0.0025 substitutions per year.}
\label{fig:backward}
\end{figure}

%%% FIGURE 4 -- SUPPLEMENT 1
\begin{figure*}
\centering
\includegraphics[width=\columnwidth]{figures/fig4_S1}
\caption{Figure 4 -- supplement 1: The distribution of plausible seeding times of reservoir HIV-1 DNA sequences for all 10 patients. For each HIV-1 DNA read obtained from the PBMCs, the plasma sample and HIV-1 RNA variant from which the read was most likely derived was determined. The plots present the distributions of these most likely origin samples across all available RNA samples.}
\label{fig:supp_timeline}
\end{figure*}

%%% FIGURE 4 -- SUPPLEMENT 2
\begin{figure*}
\centering
\includegraphics[width=\columnwidth]{figures/fig4_S2}
\caption{Figure 4 -- supplement 2: Same as Figure 4 -- supplement 1, but counting each sequence once instead of weighted by the number of reads it represents.}
\label{fig:supp_timeline_hap}
\end{figure*}

%%% SUPPLEMENTARY TABLES FOLLOW
\clearpage

%%% SEQUENCING STATISTICS
\begin{sidewaystable}[htb]
\scriptsize
\centering
\begin{tabular}{|cccc|ccc|ccc|cc|}\hline
patient & sample & \#reads & \#templates & \#seq$>0.002$ & \#seq$>0.01$ & \% recaptured & \% hyper & \#stop good& \#stop hyper& \% stop good& \%stop hyper\\ \hline
p1 & 2012-10-02 & 19363 & 820 & 58 & 27 & 33 & 13 & 8 & 1.85 & 1.4 & 83\\
p1 & 2014-10-07 & 36454 & 148 & 23 & 13 & 31 & 14 & 9 & 0.81 & 1.3 & 36\\
p1 & 2015-02-12 & 44434 & 38 & 36 & 22 & 36 & 17 & 7 & 1.43 & 0.8 & 85\\
p2 & 2015-06-09 & 16586 & 75 & 26 & 12 & NA & 29 & 9 & 1.65 & 3.9 & 99\\
p3 & 2012-09-18 & 23347 & 243 & 116 & 26 & 50 & 19 & 8 & 1.23 & 4.6 & 75\\
p3 & 2014-10-13 & 28450 & 102 & 117 & 31 & 45 & 30 & 12 & 1.78 & 4.9 & 82\\
p3 & 2015-04-24 & 44850 & 108 & 98 & 34 & 38 & 24 & 10 & 1.21 & 5.0 & 61\\
p5 & 2012-10-26 & 11190 & 180 & 23 & 14 & 14 & 21 & 7 & 1.99 & 0.9 & 94\\
p5 & 2015-03-16 & 39762 & 72 & 56 & 29 & 7 & 6 & 8 & 1.36 & 3.8 & 99\\
p6 & 2012-10-24 & 191468 & 115 & 17 & 11 & 9 & 10 & 9 & 0.94 & 1.0 & 60\\
p6 & 2014-11-03 & 33887 & 15 & 9 & 3 & 33 & 0 & 9 & nan & 1.0 & nan\\
p7 & 2014-12-01 & 35565 & 28 & 71 & 43 & 0 & 13 & 5 & 1.43 & 2.5 & 88\\
p7 & 2015-04-17 & 38323 & 108 & 67 & 41 & 0 & 11 & 5 & 1.34 & 0.5 & 94\\
p8 & 2012-09-21 & 6553 & 180 & 32 & 12 & 17 & 42 & 6 & 1.75 & 1.5 & 87\\
p8 & 2014-11-18 & 75473 & 279 & 16 & 11 & 0 & 30 & 6 & 2.00 & 0.7 & 99\\
p8 & 2015-04-07 & 18750 & 175 & 68 & 35 & 6 & 33 & 6 & 1.44 & 3.6 & 88\\
p9 & 2012-10-05 & 43306 & 60 & 64 & 43 & 19 & 13 & 5 & 0.52 & 1.1 & 44\\
p9 & 2014-10-02 & 11620 & 72 & 27 & 16 & 38 & 0 & 3 & 0.0 & 0.5 & 100\\
p9 & 2015-03-18 & 65003 & 72 & 71 & 38 & 16 & 12 & 6 & 1.86 & 0.7 & 98\\
p10 & 2012-10-09 & 7513 & 249 & 32 & 16 & 25 & 4 & 02 & 2.0 & 0.3 & 100\\
p10 & 2014-10-24 & 50537 & 116 & 74 & 29 & 28 & 14 & 16 & 1.22 & 7.1 & 99\\
p10 & 2015-02-27 & 59041 & 51 & 55 & 29 & 41 & 00 & 6 & 0.55 & 0.6 & 27\\
p11 & 2012-10-10 & 47701 & 124 & 36 & 23 & 30 & 16 & 20 & 1.00 & 6.0 & 86\\
p11 & 2014-10-22 & 32126 & 120 & 29 & 17 & 53 & 28 & 21 & 2.08 & 8.4 & 84\\
p11 & 2015-02-25 & 6834 & 123 & 57 & 37 & 35 & 33 & 11 & 0.99 & 3.7 & 75\\\hline
\end{tabular}
\begin{flushleft}
%\caption 
    {\bf Table S1. Sequencing and hypermutation statistic for all samples.}\normalsize
 
{ ``good" refers to proviral sequences that are not obviously defective, ``hyper" refers to those with an excess of \texttt{G}$\to$\texttt{A} mutations. The column ``\% recaptured" indicates the fraction of sequences $>1$\% frequency that were present in other samples from the same patient.}
\end{flushleft}
\label{tab:hypermut_stats}
\end{sidewaystable}
%%% PRIMERS
\begin{sidewaystable}[htb]
\scriptsize
\centering
\begin{tabular}{|l|l|l|}\hline
Patient/Plasmid & Primer name & Sequence \\ \hline
pAll & $\mathtt{all^{a}\_fw^{b}{1}^{c}\_689^{d}\_705^{e}}$ & \verb|ACG CAG GAC TCG GCT TGC | \\
pAll & \verb|all_fw2(nex)|${}^{f}$\verb|_770_787| & $(^{g}$\verb|TGC TCG GCA GCG TCA GAT GTG TAT AAG AGA CAG|$)^{h}$\verb|CGG AGG CTA GAA GGA GAG| \\
p1-p5, p7, p8-p11 & \verb|p1_p2_p3_p4_p5_p7_p8_p9_p10_p11_rev1_1339_1321| & \verb|AAT CTT GTG GGG TGG CTC C| \\
p6 & \verb|p6_rev1_1339_1321| & \verb|AAT CTG CTG GRG TGG CTC C| \\
p1, p7, p8, p11 & \verb|p1_p7_p8_p11_rev2(nex)_1101_1176| & \verb|(GTC TCG TGG GCT CGG AGA TGT GTA TAA GAG ACA G)GT ATA GGG TAA TTT TGG CTG| \\
p2 & \verb|p2_rev2(nex)_1191-1176| & \verb|(GTC TCG TGG GCT CGG AGA TGT GTA TAA GAG ACA G)GT ATG GGG TAA TTT TGG CTG| \\
p3 & \verb|p3_rev2(nex)_1191-1176| & \verb|(GTC TCG TGG GCT CGG AGA TGT GTA TAA GAG ACA G)GT ATC GGG TAA TTT TGG CTG| \\
p5 & \verb|p5 _rev2(nex)_1191-1176| & \verb|(GTC TCG TGG GCT CGG AGA TGT GTA TAA GAG ACA G)GT ATA GGG WAA TTT TGG CTG| \\
p6 & \verb|p6_rev2(nex)_1191-1173| & \verb|(GTC TCG TGG GCT CGG AGA TGT GTA TAA GAG ACA G)GT ATA GGA TAA TTT TGG CTG| \\
p9, p10 & \verb|p9_p10_rev2(nex)_1191-1176| & \verb|(GTC TCG TGG GCT CGG AGA TGT GTA TAA GAG ACA G)GT ATA GGG TAA TTT TGR CTG| \\
NL4-3 & \verb|NL4-3_fw1_689_705| & \verb|ACG CAG GAC TCG GCT TGC| \\
NL4-3 & \verb|NL4-3_rev1_1339_1321| & \verb|AAT CTT GTG GGG TGG CTC C| \\
NL4-3 & \verb|NL4-3_fw2(nex)_770_787| & \verb|(TGC TCG GCA GCG TCA GAT GTG TAT AAG AGA CAG) CGG AGG CTA GAA GGA GAG| \\
NL4-3 & \verb|NL4-3_rev2(nex)_1191_1173| & \verb|(GTC TCG TGG GCT CGG AGA TGT GTA TAA GAG ACA G) TAT AGG GTA ATT TTG GCT G| \\ \hline
\end{tabular}
%\caption
\normalsize
\begin{flushleft}
{{\bf Table S2. PCR primers used for HIV-1 p17gag amplification.} 
$^{a}$ Patient number. 
$^{b}$ Primer direction. 
$^{c}$ Primer position, 1=outer primer 2=inner primer. 
$^{d}$ Start position of the primer according to HXB2 coordinates. 
$^{e}$ Stop position of the primer according to HXB2 coordinates. 
$^{f}$ Nextera adapter. 
$^{g}$ Start position of the NexteraXT adapter. 
$^{h}$ Stop position of the NexteraXT adapter. 
}
\end{flushleft}
\label{tab:primers}
\end{sidewaystable}
\end{document}
